src/test/pmemset_badblock/pmemset_badblock.c

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: BSD-3-Clause
+/* Copyright 2021, Intel Corporation */
+
+/*
+ * pmemset_badblock.c -- pmemset bad block tests
+ */
+
+#include "libpmemset.h"
+#include "unittest.h"
+#include "ut_pmemset_utils.h"
+
+#include <errno.h>
+
+/*
+ * test_pmemset_src_mcsafe_badblock_read -- test mcsafe read operation with
+ *                                          encountered badblock
+ */
+static int
+test_pmemset_src_mcsafe_badblock_read(const struct test_case *tc, int argc,
+		char *argv[])
+{
+	if (argc < 1)
+		UT_FATAL("usage: test_pmemset_src_mcsafe_badblock_read <file>");
+
+	char *file = argv[0];
+	struct pmemset_source *src;
+
+	int ret = pmemset_source_from_file(&src, file);
+	UT_PMEMSET_EXPECT_RETURN(ret, 0);
+
+	size_t bufsize = 4096;
+	void *buf = MALLOC(bufsize);
```

Comment:
Shouldn't this memory be freed with `FREE` at the end of the test after using `MALLOC`?